CRM/Export/BAO/Export.php

```diff
@@ -793,125 +786,6 @@ public static function exportComponents(
         //first loop through output columns so that we return what is required, and in same order.
         $relationshipField = 0;
         foreach ($outputColumns as $field => $value) {
-          //we should set header only once
-          if ($setHeader) {
```

Comment:
Moved to public static function setHeaderRows which runs before the while loop (only once per column)